### DISCUSSION TITLE
building Jolt Physics using Emscripten

### DISCUSSION THREAD

**Discussion by geotyper** [jrouwe/JoltPhysics.js#4] [General]
If I want to compile Jolt Physics inside my c++ lib without share interfaces to JS module, is it better to take source of Jolt Physics, or try to use JoltPhysics.js. For this moment not dive into the code, so this is question about how better to start? Have experience with Bullet engine and chipmunk with Emscripten.

**Comment by jrouwe** [comment]
If you don't need the JS interface, then you don't need JoltPhysics.js either, you can just use [JoltPhysics](https://github.com/jrouwe/JoltPhysics) directly. The library supports compiling for WASM out of the box. The only thing JoltPhysics.js adds is the JS interface.

**Reply by geotyper** [reply]
Thanks for your answer. And one more small question: if looking on precompiled version there three files: jolt.js, jolt.wasm.js and jolt.wasm.wasm this is three different targets of one library or jolt.js is using them?

**Reply by jrouwe** [reply]
* jolt.js is a pure JavaScript version of Jolt
* jolt.wasm.js and jolt.wasm.wasm form a WebAssembly version of Jolt
They both offer the same interface but the WebAssembly version is much faster.
I think all main browsers support WASM but I'm not an expert on this so there may be reasons why you need the JS fallback.
